### PR TITLE
PEDS-803 filter snapshot

### PR DESCRIPTION
**Pull request by nikoferraz**

### CODE REVIEW

File: userportaldatamodel/search.py [hunk 1]
```diff
@@ -52,29 +54,44 @@ class ProjectSearch(Base):
...
+#         Integer,
+#         ForeignKey('search.id'),
 #         primary_key = True)
 
+class SearchIsShared(Base):
+    """A search that is shared with another user."""
+
+    __tablename__ = "search_is_shared"
+    search_id = Column(Integer, ForeignKey("search.id"), primary_key=True)
+    user_id = Column(
```
Comment (grugna) [inline userportaldatamodel/search.py:79]: If this is nullable, which I agree with, it shouldn't be a primary. However it should be a foreign key

File: userportaldatamodel/search.py [hunk 2]
```diff
@@ -52,29 +54,44 @@ class ProjectSearch(Base):
...
+        Integer, nullable=True, primary_key=True
+    )  # user with whom the search is shared
+    write_access = Column(Boolean, default=True)
+    delete_access = Column(Boolean, default=False)
+    shareable_token = Column(String(255), nullable=False)
+
```
Comment (grugna) [inline userportaldatamodel/search.py:85]: It misses a field to store a copy of the search

File: userportaldatamodel/search.py (hunk 1 above)
Comment (nikoferraz) [reply userportaldatamodel/search.py:79]: Right. Good point!

**Comment by grugna** [comment]
You should also add the schema for the new class/table and add the migration in the driver.py

**Comment by nikoferraz** [comment]
> You should also add the schema for the new class/table and add the migration in the driver.py

The migration is handled by Alembic now, so I don't think we want to add anything else to driver.py.

File: userportaldatamodel/search.py (hunk 2 above)
Comment (grugna) [reply userportaldatamodel/search.py:84]: @nikoferraz

File: userportaldatamodel/search.py [hunk 3]
```diff
@@ -52,29 +54,44 @@ class ProjectSearch(Base):
...
+class SearchIsShared(Base):
+    """A search that is shared with another user."""
+
+    __tablename__ = "search_is_shared"
+    search_id = Column(Integer, ForeignKey("search.id"), primary_key=True)
+    user_id = Column(
+        Integer, nullable=True, primary_key=True
+    )  # user with whom the search is shared
+    write_access = Column(Boolean, default=True)
+    delete_access = Column(Boolean, default=False)
```
[the remark this answers is not in the document]
Comment (grugna) [reply userportaldatamodel/search.py:82]: @nikoferraz

**Pull request merged**